src/main/java/bio/overture/ego/utils/Redirects.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package bio.overture.ego.utils;
+
+import static java.lang.String.format;
+
+import bio.overture.ego.model.entity.Application;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Predicate;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.security.oauth2.common.exceptions.UnauthorizedClientException;
+
+@Slf4j
+public class Redirects {
+
+  /**
+   * Returns redirect uri based on combination between provided URI and those registed in the passed
+   * application. Checks validity and throws 403 if unauthorized URI is provided.
+   *
+   * @return Returns URI as a String that Ego will redirect to
+   */
+  public static String getRedirectUri(@NonNull Application app, String redirect) {
+    val redirects = Arrays.stream(app.getRedirectUri().split(","));
+
+    // Short return if no redirect URI is provided
+    if (redirect == null || redirect.isEmpty()) {
```

Comment:
edge case : non empty, whitespace string

Reply:
Yea, I thought about this, whether or not I should do stripping after the splitting. 

My logic is if we are already in the land of comma delimited fields, it's a game of whack-a-mole with the classes of bugs that brings so I decided to not hold the hand of the user/client that sets up an application.

I can revisit this if you feel strongly otherwise.

Reply:
ignore what I just said, that is legit.

Reply:
switch to `isBlank`